modules/policy/engine/src/main/java/org/eclipse/sw360/antenna/policy/engine/EvaluationResult.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) Bosch Software Innovations GmbH 2019.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.policy.engine;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+
+/**
+ * The result object for a policy {@link Rule}. The {@link PolicyEngine} creates for each rule an object of this type
+ * and returns all {@link ThirdPartyArtifact} objects which have caused a policy violation for that rule
+ */
+public class EvaluationResult {
+    private Rule ruleForResult;
+    private Collection<ThirdPartyArtifact> failedArtifacts = new HashSet<>();
+
+    EvaluationResult(final Rule ruleForResult) {
+        this.ruleForResult = ruleForResult;
+    }
+
+    EvaluationResult(final Rule ruleForResult, final Collection<ThirdPartyArtifact> failedArtifacts) {
+        this.ruleForResult = ruleForResult;
+        this.failedArtifacts.addAll(failedArtifacts);
+    }
+
+    /**
+     * @return Business id of the associated rule
+     */
+    public String getId() {
```

Comment:
Actually, we could discuss using Kotlin in new components. I have seen code using Lombok, perhaps this would be an option, though in this case the getter allows access as facade method, is this also supported in Kotlin or Lombok?